src/udev/udevd.c

```diff
@@ -1735,6 +1691,147 @@ int main(int argc, char *argv[]) {
         if (manager)
                 udev_ctrl_cleanup(manager->ctrl);
         mac_selinux_finish();
+
+        return r;
+}
+
+static int main_forking(void) {
+        Barrier barrier = {};
+        _cleanup_(barrier_destroyp) Barrier *b = &barrier;
```

Comment:
What's that? Why not replace both lines with:
`_cleanup_(barrier_destroy) Barrier barrier = BARRIER_NULL;`

No reason to use 'barrier_destroyp'.